psutil/tests/test_process.py

```diff
@@ -799,19 +799,32 @@ def test_nice(self):
         init = p.nice()
         try:
             if WINDOWS:
-                for prio in [psutil.NORMAL_PRIORITY_CLASS,
-                             psutil.IDLE_PRIORITY_CLASS,
+                # A CI runner may limit our maximum priority, which will break
+                # this test. Instead, we test in order of increasing priority,
+                # and match either the expected value or the highest so far.
+                # We assume at least NORMAL will work, so complete failure will
+                # be caught on lower priorities.
+                highest_prio = psutil.NORMAL_PRIORITY_CLASS
```

Comment:
Actually, this needs tweaking already or the diagnostic message on failure is going to be bad.